ddtrace/contrib/pytest/plugin.py

```diff
@@ -159,10 +359,17 @@ def pytest_runtest_protocol(item, nextitem):
 
         yield
 
+    if test_suite_span is not None and (
+        nextitem is None or nextitem.parent != pytest_module_item and not test_suite_span.finished
+    ):
+        _mark_test_status(item.parent, test_suite_span)
+        test_suite_span.finish()
```

Comment:
in `flask`, it seems some suites finish before all their tests have run, so this might need some reviewing.